lib/GeoExt/data/PrintProviderBase.js

```diff
@@ -0,0 +1,797 @@
+            "_HTTPRequest": function(layer) {
+                var enc = this.encoders.layers._Layer.call(this, layer);
+                return Ext.apply(enc, {
+                    baseURL: this.getAbsoluteUrl(layer.url instanceof Array ?
+                        layer.url[0] : layer.url),
+                    opacity: (layer.opacity != null) ? layer.opacity : 1.0
+                });
+            }
+        },
+        "legends": {
+            "gx_wmslegend": function(legend, scale) {
+                var enc = this.encoders.legends.base.call(this, legend);
+                var icons = [];
+                for (var i=1, len=legend.items.getCount(); i<len; ++i) {
+                    var url = legend.items.get(i).url;
+                    if (legend.useScaleParameter === true &&
+                       url.toLowerCase().indexOf(
+                           'request=getlegendgraphic') != -1) {
+                        var split = url.split("?");
+                        var params = Ext.urlDecode(split[1]);
+                        params['SCALE'] = scale;
+                        url = split[0] + "?" + Ext.urlEncode(params);
+                    }
+                    icons.push(this.getAbsoluteUrl(url));
+                }
+                enc[0].classes[0] = {
+                    name: "",
+                    icons: icons
+                };
+                return enc;
+            },
+            "gx_wmtslegend": function(legend, scale) {
+                return this.encoders.legends.gx_urllegend.call(this, legend);
+            },
+            "gx_urllegend": function(legend) {
+                var enc = this.encoders.legends.base.call(this, legend);
+                enc[0].classes.push({
+                    name: "",
+                    icon: this.getAbsoluteUrl(legend.items.get(1).url)
+                });
+                return enc;
+            },
+            "base": function(legend){
+                return [{
+                    name: legend.getLabel(),
+                    classes: []
+                }];
+            }
+        }
+    },
+
+    /** private: property[encodersOverride]
+     *  ``Object`` Encoders override for all print content
+     */
+    encodersOverride: {
+        "layers": {
+        },
+        "legends": {
+        }
+    },
+
+    /** public: method[supportProgress]
+     *  Return true if it support progress
+     */
+    supportProgress: function() {
+        return false;
+    },
+
+    /** private: method[setAttributes]
+     */
+    setAttributes: function(attributes) {
+        this.attributes = attributes;
+        this.typesAttributes = {};
+        Ext.each(attributes, function(attribute) {
+            if (!this.typesAttributes[attribute.type]) {
+                this.typesAttributes[attribute.type] = [];
+            }
+            this.typesAttributes[attribute.type].push(attribute);
+        }, this);
+    },
+
+    /** public: method[getAttributes]
+     *  Get the current attributes.
```

Comment:
Also useful to be used externally, but it's so simple than we can remove it, as you want...